qdl: update to the latest upstream

ricardosalveti opened the pull request:
Update to the latest upstream repository, which is now at
https://github.com/linux-msm/qdl.git.

Relevant changes from latest upstream:
- USBDEVFS interface replaced with libusb
- Add ramdump support
- New utility 'ks' which uses sahara to load images from host to a
  device (relevant for Qualcomm Cloud AI 100).

Drop 0001-Makefile-Use-pkg-config-for-libxml2-detection.patch as these
changes are now available upstream.

lumag reviewed (changes requested):
Fails for me:

```
ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: QA Issue: File /usr/bin/ks in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
File /usr/bin/qdl in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
File /usr/bin/qdl-ramdump in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?) [ldflags]
ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: Fatal QA errors were found, failing task.
ERROR: Logfile of failure stored in: /home/lumag/Projects/RPB/build-rpb/tmp-rpb-glibc/work/armv8a-linaro-linux/qdl/0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314/temp/log.do_package_qa.3081535
```

lumag commented:
Also there is no need to list upstream changelog in the commit message. Please drop it. Instead provide a rationality, why the package needs to be upgraded.

lumag commented on recipes-devtools/qdl/qdl_git.bb:
```diff
@@ -1,20 +1,18 @@
 
-DEPENDS = "libxml2"
-DEPENDS:append:class-target = " udev "
+DEPENDS = "libxml2 libusb1"
```
There is `-ludev` in LDFLAGS.

ricardosalveti replied:
Where?

lumag replied:
`LDFLAGS := `xml2-config --libs` -ludev`

ricardosalveti replied:
https://github.com/linux-msm/qdl/blob/master/Makefile#L5

And actual ldflags used during the build is  "-lxml2 -lusb-1.0".

lumag commented on recipes-devtools/qdl/qdl_git.bb:
```diff
@@ -23,3 +21,4 @@ do_install () {
 }
 
 BBCLASSEXTEND = "native nativesdk"
+INSANE_SKIP:${PN}-dbg += "buildpaths"
```
NAK. This hides the problem instead of fixing it.

ricardosalveti replied:
Removed, works fine now with the cflags/ldflags changes.

ricardosalveti commented:
> Fails for me:
> 
> ```
> ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: QA Issue: File /usr/bin/ks in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
> File /usr/bin/qdl in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
> File /usr/bin/qdl-ramdump in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?) [ldflags]
> ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: Fatal QA errors were found, failing task.
> ERROR: Logfile of failure stored in: /home/lumag/Projects/RPB/build-rpb/tmp-rpb-glibc/work/armv8a-linaro-linux/qdl/0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314/temp/log.do_package_qa.3081535
> ```

Didn't fail for me, but could be my settings / environment, will test on a basic / clean setup.

ricardosalveti commented:
> Also there is no need to list upstream changelog in the commit message. Please drop it. Instead provide a rationality, why the package needs to be upgraded.

Sure, this is a common practice on other layers, which is why I added, but will remove.

ricardosalveti commented:
CI failures not related to this change.

ricardosalveti commented:
> > Fails for me:
> > ```
> > ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: QA Issue: File /usr/bin/ks in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
> > File /usr/bin/qdl in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?)
> > File /usr/bin/qdl-ramdump in package qdl doesn't have GNU_HASH (didn't pass LDFLAGS?) [ldflags]
> > ERROR: qdl-0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314-r0 do_package_qa: Fatal QA errors were found, failing task.
> > ERROR: Logfile of failure stored in: /home/lumag/Projects/RPB/build-rpb/tmp-rpb-glibc/work/armv8a-linaro-linux/qdl/0.0+b18bca7d285a3ade92b6ec94b5d41fa149658314/temp/log.do_package_qa.3081535
> > ```
> 
> Didn't fail for me, but could be my settings / environment, will test on a basic / clean setup.

I think I see what caused this, patch I dropped (which needs to be removed) changed ldflags to append instead of a force set. Will send this upstream first and get back to this.

ricardosalveti commented:
Updated and removed the local patch since it is not required anymore (after cflags/ldflags issue was merged upstream: https://github.com/linux-msm/qdl/pull/63).
